src/gluonts/model/seq2seq/_forking_network.py

```diff
@@ -231,6 +242,62 @@ def hybrid_forward(
         )
 
         return weighted_loss
+    
+    
+class ForkingSeq2SeqDistributionTrainingNetwork(ForkingSeq2SeqNetworkBase):
+    def hybrid_forward(
+        self,
+        F,
+        past_target: Tensor,
+        future_target: Tensor,
+        past_feat_dynamic: Tensor,
+        future_feat_dynamic: Tensor,
+        feat_static_cat: Tensor,
+        past_observed_values: Tensor,
+        future_observed_values: Tensor,
+    ) -> Tensor:
+        """
+        Parameters
+        ----------
+        F: mx.symbol or mx.ndarray
+            Gluon function space
+        past_target: Tensor
+            shape (batch_size, encoder_length, 1)
+        future_target: Tensor
+            shape (batch_size, encoder_length, decoder_length)
+        past_feat_dynamic
+            shape (batch_size, encoder_length, num_past_feature_dynamic)
+        future_feat_dynamic
+            shape (batch_size, encoder_length, decoder_length, num_feature_dynamic)
+        feat_static_cat
+            shape (batch_size, encoder_length, num_feature_static_cat)
+        past_observed_values: Tensor
+            shape (batch_size, encoder_length, 1)
+        future_observed_values: Tensor
+            shape (batch_size, encoder_length, decoder_length)
+
+        Returns
+        -------
+        loss with shape (batch_size, prediction_length)
+        """
+        scaled_target, target_scale = self.scaler(
+            past_target,
+            F.ones_like(past_target),  # TODO: pass the actual observed here
+        )
+        dec_output = self.get_decoder_network_output(
+            F,
+            past_target,
+            past_feat_dynamic,
+            future_feat_dynamic,
+            feat_static_cat,
+            past_observed_values,
+        )
+        distr_args = self.distr_args_proj(dec_output)
+        distr = self.distr_output.distribution(
+            distr_args, scale=target_scale.expand_dims(axis=1)
+        )
+        loss = distr.loss(future_target)
+        return loss.mean(axis = 1)
```

Comment:
The loss masking with the `future_observed_values` might be missing (see `ForkingSeq2SeqTrainingNetwork `):
```
  # mask the loss based on observed indicator
        weighted_loss = weighted_average(
            F=F, x=loss, weights=future_observed_values, axis=1
        )
```